src/main/java/org/openstreetmap/atlas/checks/validation/tag/MixedCaseNameCheck.java

```diff
@@ -0,0 +1,263 @@
+        return !(object instanceof Relation) && !this.isFlagged(object.getOsmIdentifier())
+                && ((object.getTags().containsKey(ISOCountryTag.KEY)
+                        // Must have an ISO code that is in checkNameCountries...
+                        && this.checkNameCountries
+                                .contains(object.tag(ISOCountryTag.KEY).toUpperCase())
+                        // And have a name tag
+                        && Validators.hasValuesFor(object, NameTag.class))
+                        // Or it must have a specific language name tag from languageNameTags
+                        || this.languageNameTags.stream()
+                                .anyMatch(key -> object.getOsmTags().containsKey(key)));
+    }
+
+    /**
+     * This is the actual function that will check to see whether the object needs to be flagged.
+     *
+     * @param object
+     *            the atlas object supplied by the Atlas-Checks framework for evaluation
+     * @return an optional {@link CheckFlag} object that
+     */
+    @Override
+    protected Optional<CheckFlag> flag(final AtlasObject object)
+    {
+        final List<String> mixedCaseNameTags = new ArrayList<>();
+        final Map<String, String> osmTags = object.getOsmTags();
+
+        // Check ISO against list of countries for testing name tag
+        if (this.checkNameCountries.contains(object.tag(ISOCountryTag.KEY).toUpperCase())
+                && Validators.hasValuesFor(object, NameTag.class)
+                && isMixedCase(osmTags.get(NameTag.KEY)))
+        {
+            mixedCaseNameTags.add(NameTag.KEY);
+        }
+        // Check all language name tags
+        for (final String key : this.languageNameTags)
+        {
+            if (osmTags.containsKey(key) && isMixedCase(osmTags.get(key)))
+            {
+                mixedCaseNameTags.add(key);
+            }
+        }
+
+        // If mix case id detected, flag
+        if (!mixedCaseNameTags.isEmpty())
+        {
+            this.markAsFlagged(object.getOsmIdentifier());
+            // Instruction includes type of OSM object and list of flagged tags
+            return Optional.of(this.createFlag(object,
+                    this.getLocalizedInstruction(0, object instanceof LocationItem ? "Node" : "Way",
+                            object.getOsmIdentifier(), String.join(", ", mixedCaseNameTags))));
+        }
+        return Optional.empty();
+    }
+
+    @Override
+    protected List<String> getFallbackInstructions()
+    {
+        return FALLBACK_INSTRUCTIONS;
+    }
+
+    /**
+     * Tests each word in a string for proper use of case in a name.
+     *
+     * @param value
+     *            String to check
+     * @return true when there is improper case in any of the words
+     */
+    private boolean isMixedCase(final String value)
+    {
+        // Check if it is all lower case
+        if (Pattern.compile("\\p{Lu}").matcher(value).find())
```

Comment:
So with this, and again on line 181. Regular expressions have the potential of hurting performance quite badly. Have you done any performance testing to check what kind of impact regular expression matching could have on running this particular check?